xds/src/main/java/io/grpc/xds/LocalityStore.java

```diff
@@ -379,16 +439,33 @@ private void updatePicker(
       }
 
       void shutdown() {
+        if (isDeactivated()) {
```

Comment:
It may be better to check `delayedDeletionTimer != null` directly, as it's more clear that the next line won't throw.

Reply:
Done.